.github/workflows/deploy-hubs.yaml

```
@@ -0,0 +1,85 @@
+name: Deploy and test hubs
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - deployer/**
+      - hub-templates/**
+      - requirements.txt
+      - dev-requirements.txt
+      - config/secrets.yaml
+      - config/hubs/**
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    strategy:
+      # Don't stop other deployments if one fails
+      fail-fast: false
+      matrix:
+        cluster_name:
+          # To enable auto-deployments for other clusters,
+          # add its name to the list
+          - 2i2c
+          - cloudbank
+          - carbonplan
+          - meom-ige
+          - pangeo-181919
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Check if any of our base files have changed
+        uses: dorny/paths-filter@v2
+        id: base_files
+        with:
+          filters: |
+            files:
+              - "deployer/**"
+              - "hub-templates/**"
+              - "requirements.txt"
+              - "dev-requirements.txt"
+              - "config/secrets.yaml"
+
+      - name: Check which cluster.yaml file has changed (if any)
+        uses: dorny/paths-filter@v2
+        id: config_files
+        with:
+          filters: |
+            hub_config:
+              - "config/hubs/${{ matrix.cluster_name }}.cluster.yaml"
+
+      - name: Setup gcloud
+        if: |
+          (steps.base_files.outputs.files == 'true') ||
+          (steps.config_files.outputs.hub_config == 'true')
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          version: '290.0.1'
+          # This is used for KMS only
+          project_id: two-eye-two-see
+          service_account_key: ${{ secrets.GCP_KMS_DECRYPTOR_KEY }}
+          export_default_credentials: true
```

```suggestion

      # From https://github.community/t/support-saving-environment-variables-between-steps/16230/9
      - name: Set env variable to decide whether to deploy for this cluster
        run: |
          export DEPLOY_HUB_CLUSTER = (steps.base_files.outputs.files == 'true') ||          (steps.config_files.outputs.hub_config == 'true')
          echo "DEPLOY_HUB_CLUSTER=$DEPLOY_HUB_CLUSTER" >> $GITHUB_ENV
          
      - name: Setup gcloud
        if: $DEPLOY_HUB_CLUSTER
        uses: google-github-actions/setup-gcloud@master
        with:
          version: '290.0.1'
          # This is used for KMS only
          project_id: two-eye-two-see
          service_account_key: ${{ secrets.GCP_KMS_DECRYPTOR_KEY }}
          export_default_credentials: true
```

Here's a  (probably incorrect) diff to show the kinda thing that I meant. The syntax is probably off but this is just to illustrate the idea. Does this make sense?

---

Yeah, I understand what you want to do, I just think it muddies the water by adding an extra step to traceback through (if you're debugging),  rather than clarifying things 🤷🏻‍♀️

---

ahhh I see - I didn't understand that you were talking about tracebacks. Fair enough - I don't feel strongly on this one!

---

Ha - sorry to be _that_ person, but I'm now coming around to this suggestion. More from a DRY perspective though so the condition only needs to be updated in one place as the workflow is iterated upon.